src/TLE.ts

```diff
@@ -1280,61 +1273,61 @@ export class Token {
     }
 
     public static createLeftParenthesis(startIndex: number): Token {
-        return Token.create(TokenType.LeftParenthesis, startIndex, "(");
+        return new Token(TokenType.LeftParenthesis, startIndex, "(");
     }
 
     public static createRightParenthesis(startIndex: number): Token {
-        return Token.create(TokenType.RightParenthesis, startIndex, ")");
+        return new Token(TokenType.RightParenthesis, startIndex, ")");
     }
 
     public static createLeftSquareBracket(startIndex: number): Token {
-        return Token.create(TokenType.LeftSquareBracket, startIndex, "[");
+        return new Token(TokenType.LeftSquareBracket, startIndex, "[");
     }
 
     public static createRightSquareBracket(startIndex: number): Token {
-        return Token.create(TokenType.RightSquareBracket, startIndex, "]");
+        return new Token(TokenType.RightSquareBracket, startIndex, "]");
     }
 
     public static createComma(startIndex: number): Token {
-        return Token.create(TokenType.Comma, startIndex, ",");
+        return new Token(TokenType.Comma, startIndex, ",");
     }
 
     public static createPeriod(startIndex: number): Token {
-        return Token.create(TokenType.Period, startIndex, ".");
+        return new Token(TokenType.Period, startIndex, ".");
     }
 
     public static createWhitespace(startIndex: number, stringValue: string): Token {
         assert(stringValue);
         assert(1 <= stringValue.length);
 
-        return Token.create(TokenType.Whitespace, startIndex, stringValue);
+        return new Token(TokenType.Whitespace, startIndex, stringValue);
     }
 
     public static createQuotedString(startIndex: number, stringValue: string): Token {
         assert(stringValue != null);
         assert(1 <= stringValue.length);
         assert(Utilities.isQuoteCharacter(stringValue[0]));
 
-        return Token.create(TokenType.QuotedString, startIndex, stringValue);
+        return new Token(TokenType.QuotedString, startIndex, stringValue);
     }
 
     public static createNumber(startIndex: number, stringValue: string): Token {
         assert(stringValue != null);
         assert(1 <= stringValue.length);
         assert(stringValue[0] === "-" || Utilities.isDigit(stringValue[0]));
```

Comment:
>Utilities.isDigit(stringValue[0]) [](start = 41, length = 33)

Nit: Unrelated to this PR, why check only the first character? What if this was "1z"?

Reply:
I'd have to check to be sure, but pretty sure the validation is done before calling this function.  Agreed not great design, but probably not worth the change by itself right now.